Remove conventions (#33153)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

* `CollectionUtils`

> ===== Remove deprecated custom source set interfaces

Reviewer: ```suggestion
==== Removal of `GroovySourceSet` and `ScalaSourceSet` interfaces
```

> The following source set interfaces have been removed:

Reviewer: ```suggestion
The following source set interfaces have been removed in Gradle 9.0:
```

- `org.gradle.api.tasks.GroovySourceSet`
- `org.gradle.api.tasks.ScalaSourceSet`

> Clients should configure the sources with their plugin-specific configuration:

Reviewer: ```suggestion
To configure Groovy or Scala sources, use the plugin-specific source directory set extensions:
```

- `groovy`: link:{javadocPath}/org/gradle/api/tasks/GroovySourceDirectorySet.html[GroovySourceDirectorySet]
- `scala`: link:{javadocPath}/org/gradle/api/tasks/ScalaSourceDirectorySet.html[ScalaSourceDirectorySet]

> For example, here's how you configure the groovy sources from a plugin:

Reviewer: ```suggestion
Here's how to configure Groovy sources programmatically:
```

> ```java

Reviewer: ```suggestion
[source,java]
----
```

GroovySourceDirectorySet groovySources = sourceSet.getExtensions().getByType(GroovySourceDirectorySet.class);
groovySources.setSrcDirs(Arrays.asList("sources/groovy"));

> ```

Reviewer: ```suggestion
----
```
